### DISCUSSION TITLE
When running "open-webui serve" I get an error "bad git executable"

### DISCUSSION THREAD

**Discussion by bulrush15** [open-webui/open-webui#5574] [Q&A]
This is on Windows 11 in cmd.exe prompt, using Python 3.11. I'm using the latest Brave browser v1.70.117. 

## What I did

1. Made a new project folder in my PyProjects folder for open-webui, it has an environment active. 
1. Download installer for open-webui and install it into PyProjects\openwebui\ folder. (There are no files there after I installed it.)
2. Run `open-webui serve` and I get an error "ImportError: Failed to initialize: Bad git executable." on the python line `import git`. 
3. I tried to install git with "pip install git" but the repo could not be found. The errors I got were: "ERROR: Could not find a version that satisfies the requirement git (from versions: none) ERROR: No matching distribution found for git"
4. So I downloaded and installed the git installer from https://git-scm.com/downloads. 
5. I set up my path to include "c:\program files\git\bin;C:\Users\MYUSER\Documents\PyProjects\openwebui\.venv\Scripts". I set the variable `set GIT_PYTHON_GIT_EXECUTABLE="C:\Program Files\Git\bin"` I did a test `git --version` and that worked. 
6. Running `open-webui serve` still gives me an error and I cannot browse to http://localhost:8080 or http://localhost:3000. Both addresses are in documentation. Setting `set GIT_PYTHON_GIT_EXECUTABLE="C:\Program Files\Git"` did not work either. 
7. Ollama server is running in my sys tray but I don't see any sign of the open-webui in my system tray. 

There is a lot more info on the error but the error I wrote above is the last error in the list. 

How do I get open-webui to run? 

Thank you. 

EDIT: I have set this system environment variable `GIT_PYTHON_GIT_EXECUTABLE="C:\Program Files\Git\bin"` and rebooted the PC and I still get the same error.

**Comment by Aferrenden** [comment]
I am using Miniconda for some flexibility - so I did not specifically install separate packages for python and git but ran into the same issue after the install of Open-WebUI. 

However, `git --version` did not give me any result, so the fix was a simple, Open the Anaconda Powershell Prompt and (base env) run` conda install -c anaconda git`. After the Git install, I added both user and system paths in the windows environment variables. I then created a new Miniconda environment for the Open WebUI install,` conda create -n ENVNAME python=3.11` . Activate the environment `conda activate ENVNAME `which should change your prompt, and then run` pip install open-webui` , once installed run `open-webui serve`  _problem solved._ 

Much like you, I had Ollama installed and running in Windows, and Open WebUI didn't need any additional configuration to connect to the Ollama server, picked right up and connected after install with no issue. I'm sure you may not want to go back and reconfigure, but this method is easy to implement, and keeps things separate from Ollama if you ever need to tear down and rebuild the Open WebUI environment - so I thought I would share.
